File: app/src/main/java/io/github/gmazzo/android/livewallpaper/weather/sunrisesunset/SunriseSunsetCalculator.java
package io.github.gmazzo.android.livewallpaper.weather.sunrisesunset;

import java.util.Calendar;
import java.util.TimeZone;

public final class SunriseSunsetCalculator {

    private SunriseSunsetCalculator() {
    }

    public static Calendar getSunrise(double latitude, double longitude, TimeZone timeZone, Calendar date, double degrees) {
        return new SolarEventCalculator(latitude, longitude, timeZone).computeSunriseCalendar(new Zenith(90.0d - degrees), date);
    }

    public static Calendar getSunset(double latitude, double longitude, TimeZone timeZone, Calendar date, double degrees) {
        return new SolarEventCalculator(latitude, longitude, timeZone).computeSunsetCalendar(new Zenith(90.0d - degrees), date);
    }

}

File: app/src/main/kotlin/io/github/gmazzo/android/livewallpaper/weather/sunrisesunset/SunriseSunsetCalculator.kt
package io.github.gmazzo.android.livewallpaper.weather.sunrisesunset

import java.util.Calendar
import java.util.TimeZone

object SunriseSunsetCalculator {
    fun getSunrise(
        latitude: Double,
        longitude: Double,
        timeZone: TimeZone,
        date: Calendar,
        degrees: Double
    ): Calendar? {
        return SolarEventCalculator(latitude, longitude, timeZone).computeSunriseCalendar(
            Zenith(
                90.0 - degrees
            ), date
        )
    }

    fun getSunset(
        latitude: Double,
        longitude: Double,
        timeZone: TimeZone,
        date: Calendar,
        degrees: Double
    ): Calendar? {
        return SolarEventCalculator(
            latitude,
            longitude,
            timeZone
        ).computeSunsetCalendar(Zenith(90.0 - degrees), date)
    }
}
